pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -479,11 +501,13 @@ func (s *Server) createIstioCA(opts *caOptions) (*ca.IstioCA, error) {
 	return istioCA, nil
 }
 
-func (s *Server) createSelfSignedCACertificateOptions(fileBundle *ca.SigningCAFileBundle, opts *caOptions) (*ca.IstioCAOptions, error) {
+// loadCAFromSecrets will initialize the Istio CA using istio-ca-secrets or cacerts.
+// This is only called if mounted files
+func (s *Server) loadCAFromSecrets(fileBundle *ca.SigningCAFileBundle, opts *caOptions) (*ca.IstioCAOptions, error) {
 	var caOpts *ca.IstioCAOptions
 	var err error
 	if s.kubeClient != nil {
-		log.Info("Use self-signed certificate as the CA certificate")
+		log.Info("Use cluster secret  as the CA certificate")
```

Comment:
What do you mean by cluster secret?

Reply:
A secret stored in the k8s cluster ( instead of files - which may be mounted, but also come from external secret managers). 

The main reason for Istiod to have all this code duplication with both file-based and secret is to enable the CA private key to not be stored in a Secret in the k8s cluster. If we didn't care about this case - we could as well remove the file mounts and just use the k8s Secret directly, and clean up all duplicated code paths. 

There are tools ( sidecars for Istiod - or CSI drivers ) that allow volumes to be mounted from different 'secret managers', so the private key is never accessible in the cluster - or this code path, where the private key is a cluster secret.